presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdown.java

```diff
@@ -423,6 +434,17 @@ public PlanNode visitTableScan(TableScanNode tableScan, RewriteContext<Void> con
                     tableScan.getTableConstraints(),
                     pushdownFilterResult.getLayout().getPredicate(),
                     TupleDomain.all());
+
```

Comment:
It'd be good to know when this happens. Discussed offline, we can write this as assert and see if we really hit it